hazelcast/src/main/java/com/hazelcast/internal/adapter/DataStructureAdapter.java

```diff
@@ -56,4 +56,43 @@
     LocalMapStats getLocalMapStats();
 
     boolean containsKey(K key);
+
+    /**
+     * Contains all methods of {@link DataStructureAdapter}.
+     */
+    enum DataStructureMethods implements DataStructureAdapterMethod {
```

Comment:
I've added a small and simple `DataStructureMethodsTest` which creates a runtime implemention of `DataStructureAdapter` via ByteBuddy (was already used in the project). So there is no additional test implementation which needs to be maintained when we add new methods.

I added two tests:
* all referenced methods have to be found
* all referenced methods have to have a unique signature

The second one is catching the error you've found manually:
```
java.lang.AssertionError: DataStructureAdapter::replace(Object, Object) is referenced by DataStructureMethods.REPLACE_WITH_OLD_VALUE, but was already referenced by DataStructureMethods.REPLACE
```
The first one would look like this:
```
java.lang.AssertionError: DataStructureAdapter::get(String) is referenced by DataStructureMethods.GET, but could not be found!
```